### PR TITLE
Agones Release v1.32.0

### LINKED ISSUE #3178: Release 1.32.0
## Prerequisites

- [x] Have at least `Editor` level access to `agones-images` project.
- [x] Have permission to publish new versions of the App Engine application.
- [x] Have write access to Agones GitHub repository.
- [x] Run `git remote -v` and see:
  - [x] An `origin` remote that points to a personal fork of Agones, such as `git@github.com:yourname/agones.git`.
  - [x] An `upstream` remote that points to `git@github.com:googleforgames/agones.git`.
- [x] Have a `gcloud config configurations` configuration called `agones-images` that points to the same project.
- [x] Edit access to the [Agones Release Calendar](https://calendar.google.com/calendar/u/0?cid=Z29vZ2xlLmNvbV84MjhuOGYxOGhmYnRyczR2dTRoMXNrczIxOEBncm91cC5jYWxlbmRhci5nb29nbGUuY29t)

## Steps

- [x] Run `make shell` and run `gcloud config configurations activate agones-images`.
- [x] Review merged PRs in the current milestone to ensure that they have appropriate tags.
- [x] Ensure the next stable releases in the Google Calendar have the correct version number.
- [x] `git checkout main && git pull --rebase upstream main`
- [x] Run `make pre-build-release` to ensure all example images exist on agones-images/examples repository and to deploy the 1.32.0-1 service on GCP/App Engine/Services.
- [x] Navigate to `agones` directory and run `go run build/scripts/version-update/main.go --release-stage=before 1.32.0`. This will update the version number of the Helm/SDK/Install packages to 1.32.0.
- [x] Create a _draft_ release with the [release template][release-template].
  - run `make release-example-image-markdown` to populate example images and append the output in `Images available with this release` section
  - [x] Draft a new release with [release draft][release-draft]. Update the `Tag version` and `Release title` with the release version and click on `Generate release notes` to generate the release notes with `Full Changelog` info for 1.32.0. Make sure to add the description. Include the `Images available with this release` section from the previous step that will be present after the `Full Changelog` and save the draft.
  - [x] copy the 1.32.0 release details from the `Full Changelog` and paste it on top of the CHANGELOG.md file
- [x] Site updated
  - [x] Create a new file named 1.32.0.md in `/site/content/en/blog/releases`. Copy the draft release content in this file (this will be what you send via email)[refer the previous release file].
  - run `make site-server` frequently to make sure everything looks fine for the release in your localhost
  - [x] In `site/content/en/docs/Installation/_index.md #agones-and-kubernetes-supported-versions`, for the current version, replace `{{% k8s-version %}}` with hardcoded Kubernetes versions supported by the current version. And add a row for the Agones release version with `{{% k8s-version %}}` as its supported Kubernetes versions.
  - [x] Review all `link_test` and `data-proofer-ignore` attributes and remove for link testing [delete `data-proofer-ignore` from the release file of previous version]
  - [x] Review and remove all instances of the `feature` shortcode.
    - Ignore html and release files.
    - remove the `feature expiryVersion` block with content. remove only the block of `feature publishVersion` and do not remove the content.
    - In helm.md file, merge the rows that are present in the `New Configurations Features` table into the above `Configuration` table. The `New Configurations Features` table gets left in place (but empty) and the publishVersion bumped to the next upcoming release.
  - [x] Add a link to previous version's documentation to nav dropdown in `site/layouts/partials/navbar.html`
  - [x] config.toml updates:
    - [x] Update `release_branch` to the new release branch for 1.32.0.
    - [x] Update `release_version` with the new release version 1.32.0.
    - [x] Copy `dev_supported_k8s` to `supported_k8s`.
    - [x] Copy `dev_k8s_api_version` to `k8s_api_version`.
    - [x] Copy `dev_gke_example_cluster_version` to `gke_example_cluster_version`.
    - [x] Copy `dev_aks_example_cluster_version` to `aks_example_cluster_version`.
    - [x] Copy `dev_eks_example_cluster_version` to `eks_example_cluster_version`.
    - [x] Copy `dev_minikube_example_cluster_version` to `minikube_example_cluster_version`.
    - [x] Update documentation with updated example images tags.
- [x] Create PR with these changes, and merge them with an approval.
- [x] Run `git remote update && git checkout main && git reset --hard upstream/main` to ensure your code is in line
      with upstream (unless this is a hotfix, then do the same, but for the release branch)
- [x] Publish SDK packages
  - [x] Run `make sdk-shell-node` to get interactive shell to publish node package. Requires Google internal process
        to publish.
  - [x] Run `make sdk-publish-csharp` to deploy to NuGet. Requires login credentials.
        Will need [NuGet API Key](https://www.nuget.org/account/apikeys) from Agones account.
- [x] Run `make post-build-release` to build the artifacts in GCS(These files will be attached in the release notes) and to push the latest images in the release repository.
- [x] Run `make shell` and run `gcloud config configurations activate <your development project>` to switch Agones
      development tooling off of the `agones-images` project.
- [x] Do a `helm repo add agones https://agones.dev/chart/stable` / `helm repo update` and verify that the new
      version is available via the command `helm search repo agones --versions --devel`.
- [x] Do a `helm install --namespace=agones-system agones agones/agones` or `helm install --create-namespace --namespace=agones-system agones agones/agones` if the namespace was deleted and a smoke test to confirm everything is working.
- [x] Attach all assets found in the cloud storage with 1.32.0 to the draft GitHub Release.
- [x] Copy any review changes from the release blog post into the draft GitHub release.
- [x] Publish the draft GitHub Release.
- [x] Run `make release-branch` to create a release branch.
- [x] Email the [mailing list][list] with the release details (copy-paste the release blog post).
- [ ] Paste the announcement blog post to the #users Slack group.
- [ ] Post to the [agonesdev](https://twitter.com/agonesdev) Twitter account.
- [x] Run `git checkout main`.
- [x] Then increment the `base_version` in [`build/Makefile`][build-makefile]
- [x] Navigate to `agones` directory and run `go run build/scripts/version-update/main.go --release-stage=after 1.32.0` file. This will update the version number of the Helm/SDK/Install packages to 1.32.0+1-dev.
- [ ] Create PR with these changes, and merge them with approval
- [ ] Close this issue. _Congratulations!_ - the release is now complete! :tada: :clap: :smile: :+1:

[list]: https://groups.google.com/forum/#!forum/agones-discuss
[release-template]: https://github.com/googleforgames/agones/blob/main/docs/governance/templates/release.md
[release-draft]: https://github.com/googleforgames/agones/releases
[build-makefile]: https://github.com/googleforgames/agones/blob/main/build/Makefile

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind release

**What this PR does / Why we need it**:
Release v1.32.0

**Which issue(s) this PR fixes**:

Closes #3178

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/blog/releases/1.32.0.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+---
+title: "1.32.0 -  Enhanced Game Server Allocation, Go Version Upgrade and Stable Features!"
+linktitle: "1.32.0"
+date: "2023-05-23"
+---
+
+This is the 1.32.0 release of Agones.
+
+In this release, we are pleased to announce an upgrade to Go version 1.20. This upgrade ensures that your game servers are equipped with the latest tools and improvements, enhancing their overall performance and stability. We are also promoting some features up the feature stages, which means that they are now more stable and ready for use.
```
Comment (roberthbailey) [inline site/content/en/blog/releases/1.32.0.md:9]: I think it's worth specifically calling out the features that have graduated. For an example, see https://agones.dev/site/blog/2023/01/17/1.29.0-kubernetes-1.24.0-sdkgracefultermination-feature-in-beta-new-podhostname-feature/

Comment (Kalaiselvi84) [reply site/content/en/blog/releases/1.32.0.md:9]: Does this description look good?

In this release, we are pleased to announce an upgrade to Go version 1.20. This upgrade ensures that your game servers are equipped with the latest tools and improvements, enhancing their overall performance and stability. We are also promoting ResetMetricsOnDelete, PodHostname, SplitControllerAndExtensions, SafeToEvict, and SDKGracefulTermination features up the feature stages, which means that they are now more stable and ready for use.

We encourage you to take advantage of these promoted features to enhance the stability, performance, and functionality of your game servers.

Comment (gongmax) [reply site/content/en/blog/releases/1.32.0.md]: SDKGracefulTermination is still in Beta as before, right? So it's not promoted up.

Comment (Kalaiselvi84) [reply site/content/en/blog/releases/1.32.0.md]: Yes that's correct.. I have added the below description that does not include SDKGracefulTermination feature..

This is the 1.32.0 release of Agones.

In this release, we are pleased to announce an upgrade to Go version 1.20. This upgrade ensures that your game servers are equipped with the latest tools and improvements, enhancing their overall performance and stability. We are also promoting ResetMetricsOnDelete, PodHostname, SplitControllerAndExtensions, and SafeToEvict features up the feature stages, which means that they are now more stable and ready for use.

We encourage you to take advantage of these promoted features to enhance the stability, performance, and functionality of your game servers.

Comment (roberthbailey) [reply site/content/en/blog/releases/1.32.0.md]: This upgrade ensures that your game servers are equipped with the latest tools and improvements, enhancing their overall performance and stability:

* Upgrade to Go version 1.20. 
* The `ResetMetricsOnDelete` feature is now beta and enabled by default. Now when a Fleet or FleetAutoscaler is deleted from the system, Agones will automatically clear metrics that utilize their name as a label from the exported metrics, so the metrics exported do not continuously grow in size over the lifecycle of the Agones installation.
* The `PodHostname` feature that allows you to access a GameServer Pod within a cluster by a DNS name is now beta and enabled by default.
* The `SplitControllerAndExtensions` feature which splits the Agones controller into multiple binaries for better reliability is now beta and enabled by default.
* The `SafeToEvict` and `SDKGracefulTermination` features have both graduated to stable.

We encourage you to take advantage of these promoted features to enhance the stability, performance, and functionality of your game servers.

Comment (gongmax) [reply site/content/en/blog/releases/1.32.0.md]: Just a reminder that `SDKGracefulTermination` is still in beta

File: site/content/en/docs/Guides/feature-stages.md [hunk 2]
```diff
@@ -24,30 +24,11 @@ that can be found in the [Helm configuration]({{< ref "/docs/Installation/Instal
...
-{{% /feature %}}
-
-{{% feature publishVersion="1.32.0" %}}
-
 | Feature Name                                                                                                          | Gate                           | Default  | Stage   | Since  |
 |-----------------------------------------------------------------------------------------------------------------------|--------------------------------|----------|---------|--------|
 | [Custom resync period for FleetAutoscaler](https://github.com/googleforgames/agones/issues/1955)                      | `CustomFasSyncInterval`        | Enabled  | `Beta`  | 1.25.0 |
 | [GameServer Stable Network ID]({{% ref "/docs/Reference/gameserver.md#stable-network-id" %}})                         | `PodHostname`                  | Enabled  | `Beta`  | 1.32.0 |
-| [Reset Metric Export on Fleet / Autoscaler deletion]({{% relref "./metrics.md#dropping-metric-labels" %}})            | `ResetMetricsOnDelete`         | Enabled  | `Beta`  | 1.26.0 |
+| [Reset Metric Export on Fleet / Autoscaler deletion]({{% relref "./metrics.md#dropping-metric-labels" %}})            | `ResetMetricsOnDelete`         | Enabled  | `Beta`  | 1.32.0 |
```
Comment (roberthbailey) [inline site/content/en/docs/Guides/feature-stages.md:31]: great catch!

**Review by roberthbailey — APPROVED** [review]

**Pull request merged**